bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/overheads/Overheads.scala

```diff
@@ -0,0 +1,130 @@
+package edu.gemini.model.p1.overheads
+
+import java.time.Duration
+
+import edu.gemini.model.p1.immutable._
+
+import scalaz.Scalaz._
+
+// This is not the ideal package for this to live in, but to avoid circular bundle references, we put it here.
+sealed trait Overheads {
+  def partnerOverheadFraction: Double
+  def acquisitionOverhead: Duration
+  def typicalExpTime: Option[Duration]
+  def otherOverheadFraction: Double
+
+  // Careful here: if we convert to hours directly using java.time API, this will be a Long value representing
+  // the floor of the number of hours.
+  private def acqOverheadHrs = acquisitionOverhead.toMinutes.toDouble / 60.0
+
+  def calculate(intTime: TimeAmount): ObservationTimes = {
+    // Calculate everything in hours, as per REL-2985 formulae.
+    val progTimeHrs  = {
+      val visTimeHrs = Overheads.visTimeHrs
+      val intTimeHrs = intTime.toHours.value
+      val overheadTimeHrs = (1 + otherOverheadFraction) * intTimeHrs
+      val numAcqs = ((overheadTimeHrs / visTimeHrs).toInt + ((overheadTimeHrs % visTimeHrs > 0) ? 1 | 0)).toDouble
+
+      // Careful here: if we convert to hours directly using java.time API, this will be a Long value representing
+      // the floor of the number of hours.
+      val acqOverheadHrs = acquisitionOverhead.toMinutes.toDouble / 60.0
+      numAcqs * acqOverheadHrs + overheadTimeHrs
+    }
+    val partTimeHrs  = progTimeHrs * partnerOverheadFraction
+    val totalTimeHrs = progTimeHrs + partTimeHrs
+
+    ObservationTimes(
+      TimeAmount(progTimeHrs,  TimeUnit.HR),
+      TimeAmount(partTimeHrs,  TimeUnit.HR),
+      TimeAmount(totalTimeHrs, TimeUnit.HR))
+  }
+
+  // This is needed for 2017A to 2017B migration, as time in former proposals will be migrated to progTime, and
+  // thus intTime - the value that is user modifiable - must be calculated from this.
+  def intTimeFromProgTime(progTime: TimeAmount): TimeAmount = {
+    val visTimeHrs  = Overheads.visTimeHrs
+    val progTimeHrs = progTime.toHours.value
+    val numAcqs     = ((progTimeHrs / visTimeHrs).toInt + ((progTimeHrs % visTimeHrs > 0) ? 1 | 0)).toDouble
```

Comment:
Maybe I'm just being completely blind / stupid here, but I don't immediately see the duplication?

Are you referring to having both `intTime` and `progTime`? Again, this comes from the migration from older proposals. It will result in `progTime` being used to calculate `intTime`, which is then used to (again) calculate `progTime` and `partTime`, which is a bit redundant, but my concern was with the way these were being formatted in the UI and precision. Maybe that isn't an issue and we don't require the roundabout computation.